ui/components/app/signature-request/signature-request.component.js

```diff
@@ -116,16 +122,28 @@ export default class SignatureRequest extends PureComponent {
             {this.context.t('sigRequest')}
           </div>
           <div className="signature-request-content__identicon-container">
-            <div className="signature-request-content__identicon-initial">
-              {domain.name && domain.name[0]}
-            </div>
+            {targetSubjectMetadata?.iconUrl ? (
+              <SiteIcon
+                icon={targetSubjectMetadata.iconUrl}
+                name={
+                  getURLHostName(targetSubjectMetadata.origin) ||
+                  targetSubjectMetadata.origin
+                }
+                size={24}
+              />
+            ) : (
+              <div className="signature-request-content__identicon-initial">
+                {/* {domain.name && domain.name[0]} */}
+                {originHostNameCharacter}
+              </div>
+            )}
             <div className="signature-request-content__identicon-border" />
             <Identicon address={fromAddress} diameter={70} />
```

Comment:
I don't think this identicon should be here. This represents the signing account, but it's not being shown in a way that communicates to the user which account is being used to sign. The signing account is already being shown in the top-left corner (with both the identicon and name)

Reply:
Fixed